art/estimators/classification/pytorch.py

```diff
@@ -1208,3 +1208,85 @@ def get_layers(self) -> List[str]:
 
         except ImportError:  # pragma: no cover
             raise ImportError("Could not find PyTorch (`torch`) installation.") from ImportError
+
+
+class PyTorchClassifierDeiT(PyTorchClassifier):
+    """
+    This class implements a ViT classifier with the PyTorch framework.
+    """
+
+    def __init__(
+        self,
+        model: "torch.nn.Module",
+        loss: "torch.nn.modules.loss._Loss",
+        input_shape: Tuple[int, ...],
+        nb_classes: int,
+        optimizer: Optional["torch.optim.Optimizer"] = None,  # type: ignore
+        use_amp: bool = False,
+        opt_level: str = "O1",
+        loss_scale: Optional[Union[float, str]] = "dynamic",
+        channels_first: bool = True,
+        clip_values: "CLIP_VALUES_TYPE" = (0.0, 1.0),
+        preprocessing_defences: Union["Preprocessor", List["Preprocessor"], None] = None,
+        postprocessing_defences: Union["Postprocessor", List["Postprocessor"], None] = None,
+        preprocessing: "PREPROCESSING_TYPE" = (0.0, 1.0),
+        device_type: str = "gpu",
+    ) -> None:
+        """
+        TODO
+        """
+
+        super().__init__(
+            model=model,
+            clip_values=clip_values,
+            channels_first=channels_first,
+            preprocessing_defences=preprocessing_defences,
+            postprocessing_defences=postprocessing_defences,
+            preprocessing=preprocessing,
+            device_type=device_type,
+            nb_classes=nb_classes,
+            input_shape=input_shape,
+            loss=loss,
+            optimizer=optimizer,
+            use_amp=use_amp,
+            opt_level=opt_level,
+            loss_scale=loss_scale,
+        )
+
+    @property
+    def patch_size(self):
+        """
+        TODO
+        """
+        return self.model.patch_embed.patch_size[0]
+
+    def get_attention_weights(self, x: Union[np.ndarray, "torch.Tensor"]) -> "torch.Tensor":
```

Comment:
I think this method could be of interest for other models too. Please move it to `PyTorchEstimator` and generalise it by making `return_nodes` a list of strings provided by the user as an argument.